### ISSUE TITLE
Should an operator method that returns `Never` indicate an error?

### ISSUE THREAD

**Issue by Dr-Irv** [microsoft/pyright#4430]
See https://github.com/python/mypy/issues/14424

**Describe the bug**
Open question as to whether a subclass that returns `Never` for an operator should be flagged as incorrect code.  See discussion in mypy issue.

**To Reproduce**
```python
from __future__ import annotations
from typing import Any
from typing_extensions import Never, assert_type


class Base:
    def __add__(self, other: Any) -> Base:
        return Base()


class Sub:
    def __add__(self, other: Any) -> Never:
        pass


b = Base()
s = Sub()
br1 = b + 10
s + 10
```

pyright reports on the last line: `Operator "+" not supported for types "Sub" and "Literal[10]"`

Question is whether that is right or wrong.  mypy people say it is wrong.

**Expected behavior**
Unclear.

**VS Code extension or command-line**
pyright command line 1.1.286
